### ISSUE TITLE
What's the difference between lit-gpt and transformer module  in huggingface ?

### ISSUE THREAD

**Issue by imanu20** [Lightning-AI/lit-gpt#776]
by using huggingface transformer, people can easily write LLM pretrain project, so what is the diffence between lit-gpt and huggingface transformer module?

**Comment by Andrei-Aksionov** [comment]
> [!IMPORTANT]
> This is my personal opinion. @awaelchli @carmocca @lantiga might have a different one.

If I understood you correctly, you are wondering what is the purpose of Lit-GPT when HF Transformers exists.

**Lit-GPT: Hackable implementation of state-of-the-art open-source LLMs.**

Let me unwrap what it means.

It means that this repo is a perfect choice for anyone who wants to experiment with modern LLMs, understand and extend them, as the code is simple (*Single-file implementation without boilerplate*), without many layers of abstraction. 

For example, let's say you read a paper about a simple technique to improve positional encodings and want to give it a try. You want an existing codebase that is simple, supports all the latest LLM architectures, so you can figure out in no time where you should apply your changes and quickly see the result. This repo is your choice.

Or, for educational purposes, you want to see a simple implementation of, I don't know, LoRA (I'm a bit biased here 😄).  Again, Lit-GPT is to the rescue.

In addition, it's powered by **Fabric**, which makes the code agnostic to hardware, precision and distribution backend. So switching between those things is trivial and doesn't require adding boilerplate code (😉 Adrian).

Don't forget about [Discord](https://discord.gg/VptPCZkGNa) channel, where you can get answers to your questions.

-----

*In a nutshell, as I see it, this repo is a perfect starter-kit for an LLM project of any scale. A sound foundation that can be integrated, modified and extended with as less friction as possible.*

-----

@imanu20 Does it answer your question?
